src/utils.ts

```diff
@@ -83,9 +85,17 @@ export function getFloat(_value0: number, _value1: number) {
 	return parseFloat(ldexp(0.01 * mantissa, exponent).toPrecision(15))
 }
 
-export function wait(ms: number) {
-	return new Promise<void>((r) => {
-		setTimeout(r, ms)
+export function wait(ms: number): Promise<void> {
+	return new Promise<void>((resolve) => {
+		const start = performance.now()
+		function check() {
+			if (performance.now() - start >= ms) {
+				resolve()
+			} else {
+				setImmediate(check) // Not to block the event loop
+			}
+		}
+		check()
 	})
 }
```

Comment:
previously setTimeout wait, on slow server (such as old raspberryPI compute modules), is not accurate. For example, a pause of 30ms becomes much longer. I'm trying other ways to do a wait that better respects the interval.
Do you have other good ideas?

Reply:
Yeah but in this way you are nuking the js stack trace. Maybe this:

```ts
function sleep(ms) {
  return new Promise((resolve) => {
    const start = process.hrtime.bigint();
    const end = start + BigInt(ms) * BigInt(1e6); // Convert ms to nanoseconds

    function checkTime() {
      const now = process.hrtime.bigint();
      if (now >= end) {
        resolve();
      } else {
        setTimeout(checkTime, Math.min(Number(end - now) / 1e6, 1)); // Check again in 1 ms or less
      }
    }

    setTimeout(checkTime, Math.min(ms, 1)); // Initial check
  });
}
```

Didn't tested

Reply:
Otherwise this is the best alternative: https://github.com/Krb686/nanotimer

Reply:
Nanotimer uses  process.hrtime. The problem is, that since nodejs has an event loop that can be freezed by an asynchronous call. We have not so many choices here.